File: api/Reactello.Domain/Entities/Board.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reactello.Domain.Entities
{
    public class Board : EntityBase
    {
        public Board()
        {
            Sections = new HashSet<Section>();
        }
        public bool IsPublic { get; set; }
        public string Name { get; set; }
        public string Background { get; set; }
        public ICollection<Section> Sections { get; private set; }
    }
}

File: src/models/board.ts
import { NormalizedBoard, NormalizedModel } from './normalizedModels';
import { Section } from './section';

export interface BoardForm {
	id?: number;
	isPublic: boolean;
	name: string;
	background: string;
}

export interface Board {
	id: number;
	isPublic: boolean;
	name: string;
	background: string;
	dateCreated?: string;
	sections: Section[];
}

export const normalize = (board: Board): NormalizedModel<NormalizedBoard> => {
	return {
		[board.id]: {
			id: board.id,
			isPublic: board.isPublic,
			name: board.name,
			background: board.background,
			dateCreated: board.dateCreated,
			sections: board.sections.map((x) => x.id)
		}
	};
};
